openedx/core/djangoapps/content_libraries/api/libraries.py

```diff
@@ -229,24 +281,40 @@ class CollectionMetadata:
 
 
 @dataclass(frozen=True)
-class LibraryXBlockMetadata:
+class LibraryItem:
     """
-    Class that represents the metadata about an XBlock in a content library.
+    Common fields for anything that can be found in a content library.
     """
-    usage_key: LibraryUsageLocatorV2
     created: datetime
     modified: datetime
+    display_name: str
+
+
+@dataclass(frozen=True, kw_only=True)
+class PublishableItem(LibraryItem):
+    """
+    Common fields for anything that can be found in a content library that has
+    draft/publish support.
+    """
     draft_version_num: int
     published_version_num: int | None = None
-    display_name: str = ""
     last_published: datetime | None = None
-    # THe username of the user who last published this.
+    # The username of the user who last published this.
     published_by: str = ""
     last_draft_created: datetime | None = None
     # The username of the user who created the last draft.
     last_draft_created_by: str = ""
     has_unpublished_changes: bool = False
     collections: list[CollectionMetadata] = field(default_factory=list)
+    can_stand_alone: bool = True
```

Comment:
I don't think we should expose `can_stand_alone` on the dataclasses -- it's [for internal use only](https://github.com/openedx/openedx-learning/pull/289#issuecomment-2737787365).

Reply:
See below: https://github.com/openedx/edx-platform/pull/36434#discussion_r2015862291, we need the frontend to set this flag to `False` whenever a new component is created under a container like unit.

Reply:
Ah ok -- I missed that subtlety, that's fine :)

But could you add some documentation about the parameters to `create_library_block` to explain what this flag is for to users of the API?

Reply:
@pomegranited Done. cb9ea4fe64db